lib/fprotect/fprotect_rram.c

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdint.h>
+#include <string.h>
+#include <errno.h>
+#include <soc.h>
+#include <nrf_erratas.h>
+#include <hal/nrf_rramc.h>
+
+#define RRAMC_REGION_FOR_FPROTECT 4
+#define REGION_SIZE_MAX ((RRAMC_REGION_CONFIG_SIZE_Msk >> RRAMC_REGION_CONFIG_SIZE_Pos) * CONFIG_NRF_RRAM_REGION_SIZE_UNIT)
+
+int fprotect_area(uint32_t start, size_t size)
+{
+	if( start % CONFIG_NRF_RRAM_REGION_ADDRESS_RESOLUTION ||
+	    size % CONFIG_NRF_RRAM_REGION_SIZE_UNIT ) {
+		/*
+		 * We avoid setting up locks from address lower than requested
+		 * which occurs due to difference in resolution.
+		 * Also we don't want to lock more than requested.
+		 */
+		return -EINVAL;
+	}
+	if (size <= REGION_SIZE_MAX) {
```

Comment:
You should also check if RRAMC_REGION_FOR_FPROTECT is already in use. If it is, return -NOSPC